src/celldega/nbhd/__init__.py

```diff
@@ -205,3 +213,190 @@ def alpha_shape_geojson(gdf_alpha, meta_cluster, inst_alpha):
+
+    return gdf_hexatile
+
+def generate_random_points_gdf(n_points=10, x_range=(0, 100), y_range=(0, 100)):
+    points = [
+        Point(random.uniform(*x_range), random.uniform(*y_range))
+        for _ in range(n_points)
+    ]
+    gdf_trx = gpd.GeoDataFrame(geometry=points)
+    return gdf_trx
+
+def generate_dummy_image(width=100, height=100):
+
+    data = np.random.randint(0, 256, (height, width, 3), dtype=np.uint8)
+    img = Image.fromarray(data, 'RGB')
+    img_np = np.array(img)
+    return img_np
+
+def hexatile_area(gdf_hexatile, tissue_img_shape):
+
+    return round(sum(gdf_hexatile.area.tolist())) == round(tissue_img_shape[0] * tissue_img_shape[1])
+
+def hexatile_trx_assignment(hexatile_assigned_trx):
+
+    more_than_one_matches = 'more_than_one_matches' in hexatile_assigned_trx['polygon_index'].unique()
+    unassigned = 'UNASSIGNED' in hexatile_assigned_trx['polygon_index'].unique()
+
+    return more_than_one_matches, unassigned
+
+def unassigned_transcripts_tiled_view(gdf_hexatile, path_default_data, path_landscape_files, path_segmentation_files=None):
+
+    from ..pre.merge_segmentations import assigning_transcripts
+
+    transformation_matrix = pd.read_csv(os.path.join(path_landscape_files, "micron_to_image_transform.csv"), sep=" ", header=None).values[:3,:3]
+
+    with open(os.path.join(path_landscape_files, 'landscape_parameters.json'), 'r') as landscape_parameters_file:
+        landscape_parameters = json.load(landscape_parameters_file)
+
+    if landscape_parameters['technology'] == 'Xenium':
+        x = 'x_location'
+        y = 'y_location'
+
+    elif landscape_parameters['technology'] == 'MERSCOPE':
+        x = 'center_x'
+        y = 'center_y'
+
+    else:
+        x = 'x'
+        y = 'y'
+
+    print("Reading transcripts file...")
+    if isinstance(path_segmentation_files, str):
+        trx = pd.read_parquet(os.path.join(path_segmentation_files, "transcripts.parquet"))
+
+    else:
+        trx = pd.read_parquet(os.path.join(path_default_data, "transcripts.parquet"))
+
+    print("Transcripts file read.")
+
+    gdf_trx = gpd.GeoDataFrame(trx, geometry=gpd.points_from_xy(trx[x], trx[y]))
+    gdf_trx['geometry_image_space'] = gdf_trx['geometry'].apply(lambda geom: affine_transform(geom, [transformation_matrix[0, 0],
+                                                                                               transformation_matrix[0, 1],
+                                                                                               transformation_matrix[1, 0],
+                                                                                               transformation_matrix[1, 1],
+                                                                                               transformation_matrix[0, 2],
+                                                                                               transformation_matrix[1, 2]]))
+
+    gdf_trx.set_geometry('geometry_image_space', inplace=True)
```

Comment:
The `unassigned_transcripts_tiled_view()` step is pretty low on my end. I guess the `apply` with `affine_transform` is the slow part as it is not a vectorized operation. Can you try something like below that I used in the codebase and that served me well? 

```
from ..pre.boundary_tile import batch_transform_geometries
from ..pre.__init__ import _to_geometry
gdf_trx["geometry"] = batch_transform_geometries(gdf_trx["geometry"], transformation_matrix, 1)
gdf_trx['geometry'] = gdf_trx['geometry'].apply(_to_geometry)
```